lib/providers/app_authentication/app_authentication_providers.dart

```diff
@@ -0,0 +1,264 @@
+import "dart:async";
+import "dart:developer";
+
+import "package:android_id/android_id.dart";
+import "package:flutter_appauth/flutter_appauth.dart";
+import "package:flutter_secure_storage/flutter_secure_storage.dart";
+import "package:riverpod_annotation/riverpod_annotation.dart";
+import "package:vp_kuljetus_driver_app/app/env.gen.dart";
+import "package:vp_kuljetus_driver_app/models/authentication/authentication.dart";
+import "package:vp_kuljetus_driver_app/services/api/api.dart";
+import "package:vp_kuljetus_driver_app/services/store/store.dart";
+
+part "app_authentication_providers.g.dart";
+
+const serviceConfiguration = AuthorizationServiceConfiguration(
+  authorizationEndpoint:
+      "${Env.keycloakUrl}/realms/${Env.keycloakRealm}/protocol/openid-connect/auth",
+  tokenEndpoint:
+      "${Env.keycloakUrl}/realms/${Env.keycloakRealm}/protocol/openid-connect/token",
+  endSessionEndpoint:
+      "${Env.keycloakUrl}/realms/${Env.keycloakRealm}/protocol/openid-connect/logout",
+);
+
+@Riverpod(keepAlive: true)
+class AppAuthNotifier extends _$AppAuthNotifier {
+  late final StreamController<AuthenticationState?> _controller;
+  final FlutterAppAuth _appAuth = const FlutterAppAuth();
+  final FlutterSecureStorage _secureStorage = const FlutterSecureStorage();
+  Timer? _refreshTimer;
+
+  @override
+  Stream<AuthenticationState?> build() {
+    _controller = StreamController<AuthenticationState?>();
+    state = const AsyncData(null);
+    ref.onDispose(() {
+      _controller.close();
+      _refreshTimer?.cancel();
+    });
+    return _controller.stream;
+  }
+
+  void _stopRefreshTimer() {
+    _refreshTimer?.cancel();
+    _refreshTimer = null;
+  }
+
+  Future<AuthenticationState> _doCommonLoginActions(
```

Comment:
Maybe more descriptive?

```suggestion
  Future<AuthenticationState> _handleSuccessfulLoginResponse(
```